src/analyzer.ts

```diff
@@ -203,16 +201,47 @@ export class Analyzer {
               col: scannedDependency.locationOffset.col,
               filename: document.url
             };
-            const scannedDocument = await this._analyzeSource(
-                scannedDependency.type, scannedDependency.contents,
-                document.url, locationOffset,
-                scannedDependency.attachedComment);
-            scannedDependency.scannedDocument = scannedDocument;
-            scannedDependency.scannedDocument.isInline = true;
-            return scannedDocument;
+            try {
+              const scannedDocument = await this._analyzeSource(
```

Comment:
how about one method to analyze an `InlineParsedDocument` and another to analyze `ScannedImport`. The methods name themselves even.